src/Raven.Server/Smuggler/Documents/StreamSource.cs

```diff
@@ -1758,9 +1783,122 @@ private async IAsyncEnumerable<DocumentItem> ReadDocumentsAsync(List<string> col
+                    if (_buildVersionType < BuildVersionType.V7)
+                    {
+                        if (metadata.TryGet(Constants.Documents.Metadata.Attachments, out BlittableJsonReaderArray att) && att.Length > 0)
+                        {
+                            var didWork = false;
+                            var results = new DynamicJsonArray();
+                            for (var i = 0; i < att.Length; i++)
+                            {
+                                var attachmentInMetadata = (BlittableJsonReaderObject)att[i];
+                                if (attachmentInMetadata.TryGet(nameof(AttachmentName.Hash), out string hash) == false)
+                                {
+                                    // this should skip this attachment from the results array
+                                    didWork = true;
+                                    if (_log.IsInfoEnabled)
+                                        _log.Info($"Ignoring an attachment because couldn't parse its hash: {attachmentInMetadata}");
+                                    continue;
+                                }
+
+                                if (attachmentInMetadata.TryGet(nameof(AttachmentName.Size), out long _) == false)
+                                {
+                                    if (hashBySize.ContainsKey(hash) == false)
+                                    {
+                                        if (_log.IsInfoEnabled)
+                                            _log.Info(
+                                                $"Ignoring an attachment with hash '{hash}' for document '{modifier.Id}' because couldn't find its size. Attachment metadata: {attachmentInMetadata}");
+                                        Debug.Assert(false, "imported attachment doesn't have size & stream");
+                                        attachmentInMetadata.Modifications = new DynamicJsonValue(attachmentInMetadata) { [nameof(AttachmentName.Size)] = 0L };
+                                    }
+                                    else
+                                    {
+                                        attachmentInMetadata.Modifications = new DynamicJsonValue(attachmentInMetadata)
+                                        {
+                                            [nameof(AttachmentName.Size)] = hashBySize[hash]
+                                        };
+                                    }
+                                }
+
+                                if (attachmentInMetadata.TryGet(nameof(AttachmentName.Flags), out AttachmentFlags flags) == false)
+                                {
+                                    flags |= AttachmentFlags.None;
+                                    if (attachmentInMetadata.Modifications == null)
+                                    {
+                                        attachmentInMetadata.Modifications = new DynamicJsonValue(attachmentInMetadata)
```

Comment:
this is repeated few times, can you wrap it up in a method to reduce number of code here?

Reply:
done